### DISCUSSION TITLE
Impostazione retention

### DISCUSSION THREAD

**Discussion by marcomeneghin** [globaleaks/globaleaks-whistleblowing-software#3436] [Q&A]
Buongiorno,
  non capiamo come attivare la retention delle segnalazioni. In particolare volevamo che le segnalazioni chiuse oltre due anni venissero automaticamente cancellate dal sistema per adempiere alle previsioni del GDPR.

Grazie, M.

**Comment by evilaliv3** [comment]
Buongiorno @marcomeneghin,

al momento globaleaks prevede la possibilità di configurare una data retention policy per ogni canale di segnalazione.

Se non hai al momento configurato più di un contesto di segnalazione trovi questa impostazione dentro al canale di defaul chiamato "Default".

E' stato ipotizzato che in futuro si imlementi delle policy per stato così che variando lo stato di segnalazione si possa inserire una deroga ma per il momento i nostri attuali contributors sono impegnati su altri fronti.

**Reply by rlovisetto** [reply]
Buongiorno Giovanni,
mi inserisco in questa discussione per chiedere se il periodo di "retention" è gestito anche per i LOG o se invece gestisce solo le SEGNALAZIONI. Abbiamo visto che nella configurazione è possibile inserire il termine del periodo per le segnalazioni ma non  abbiamo trovato nulla circa la possibilità di definire il periodo per i LOG. Ci puoi, per favore, aiutare a capire questo aspetto?

**Comment by marcomeneghin** [comment]
Buongiorno Giovanni,
  grazie delle info ma dentro i contesti non trovo il parametro:

***@***.***

Dove dovrebbe essere?

Noi stiamo utilizzando questa Versione del software: 4.10.18, è per questo che non vedo l’opzione?

Grazie, Marco.

Da: Giovanni Pellerano ***@***.***>
Inviato: venerdì 28 aprile 2023 08:40
A: globaleaks/GlobaLeaks ***@***.***>
Cc: Marco Meneghin ***@***.***>; Mention ***@***.***>
Oggetto: Re: [globaleaks/GlobaLeaks] Impostazione retention (Discussion #3436)


Buongiorno @marcomeneghin<https://github.com/marcomeneghin>,

al momento globaleaks prevede la possibilità di configurare una data retention policy per ogni canale di segnalazione.

Se non hai al momento configurato più di un contesto di segnalazione trovi questa impostazione dentro al canale di defaul chiamato "Default".

E' stato ipotizzato che in futuro si imlementi delle policy per stato così che variando lo stato di segnalazione si possa inserire una deroga ma per il momento i nostri attuali contributors sono impegnati su altri fronti.

**Comment by evilaliv3** [comment]
Grazie per la domanda.

I log implementati sono privacy preserving e Al momento tenuti
indefinitamente fino a cancellazione dell'istanza in caso di multitenancy.

Per ragioni non tanto di compliance quanto di risparmio si prevede che
probabilmente dovranno essere implementate cancellazioni ad esempio
predisponendo che i log vengano mantenuto per tot anni configurabili
rispetto al tipo di log.

Avete raccolto qualche necessità specifica o identificato qualche necessità
ai sensi di legge?

**Reply by rlovisetto** [reply]
Grazie per la pronta risposta.

Il Regolamento Generale sulla Protezione dei Dati (GDPR) non fornisce linee guida specifiche sulla retention dei file di log. Tuttavia, il principio generale della minimizzazione dei dati, incluso nella GDPR, implica che i dati personali debbano essere conservati solo per il tempo necessario agli scopi per i quali sono stati raccolti. Ciò significa che anche i file di log contenenti dati personali devono essere conservati solo per un periodo di tempo ragionevole.

I file log accessibili all'utente amministratore contengono i log degli utenti con l'indicazione di tutti i loro accessi. Tali dati, per come ci è stato rappresentato dai nostri consulenti legali, rappresentano dati personali.

Avete previsto delle tempisiche in merito alle implementazioni citate nella risposta?
Grazie

**Comment by evilaliv3** [comment]
Non abbiamo previsto tempistiche ma diciamo che è una funzionalità semplice da implementarsi, va capito giusto cosa permettere di fare.

I vostri consulenti cosa suggeriscono?

1. Basta una policy unica che dica cancella i dati dopo tot tempo?
2. Questo tot tempo a vostro parere quanto dovrebbe essere? A mio avviso stante che la legge whistleblowing consente la conservazione fino a 5 anni per le segnalazioni dal momento dell'esito, ritengo adeguato mantenere i log per 5 anni che è compatibile con il GDPR in quanto la finalità di audit su un processo così critico ne giustifica a mio avviso tale durata di trattamento.

**Reply by rlovisetto** [reply]
I nostri consulenti suggeriscono  di differenziare il periodo di retention, Per i LOG suggeriscono 12 mesi per le segnalazioni va giò bene come adesso in quanto ognuno può decidere il periodo di retention.

**Comment by marcomeneghin** [comment]
Salve, non ho più ricevuto risposta in merito a dove vanno impostati i critery di retention dei casi. Ricordo che la normativa italiana parla chiaro in merito:
Art. 14 Conservazione della documentazione inerente alle segnalazioni 1. Le segnalazioni, interne ed esterne, e la relativa documentazione sono conservate per il tempo necessario al trattamento della segnalazione **e comunque non oltre cinque anni a decorrere dall adata della comunicazione dell'esito finale della procedura di segnalazione,** nel rispetto degli obblighi di riservatezza di cuiall'articolo 12 del presente decreto e del principio di cui agliarticoli 5, paragrafo 1, lettera e), del regolamento (UE) 2016/679 e3, comma 1, lettera e), del decreto legislativo n. 51 del 2018.
Grazie, M.

**Comment by evilaliv3** [comment]
Buonasera Marco,

la ringrazio per la domanda.
Le policy di data retention sono impostabili sui contesti di segnalazione dove trova impostabile un peridio di scadenza.

https://docs.globaleaks.org/en/main/user/admin/UserInterface.html#submissions-expiration

Il software imposta questa tempistica a 90 giori
ni per ispirare buone pratiche. Lo porteremo con il prossimo aggiornamento a 180 giorni che è tempistica ragionevole in contesto europeo dato che il trattamento di una segnalazione va concluso a 90 giorni e ragionevolmente è adeguato mantenere per un tempo similare la segnalazione a sistema per dare possibilità ai segnalanti.

Questo consente secondo una impostazione privacy by default e privacy by design di fare scadere tutto con una tempisitca certa impostata sui tempi di gestione. Sono poi i riceventi che se necessario possono di fatto tenere le segnalazioni per più tempo eventualmente fino a 5 anni secondo legge italiana ma quello è un loro adempimento.

Prevediamo poi di aggiungere qualche impostazione di data retention anche collegata agli stati di segnalazione ma questa è una impostazione che ancora necessità di appropriata ricerca di usabilità e risorse di sviluppo . La funzionalità ptra magari aiutare per ciò che riferisce ma consideriamo che porti al rischio di realizzare implementazioni dove tutto venga sempre tenuto 5 anni, che è contro il GDPR ad il buon senso di responsabilità.

Cosa ne pensa? Mi chieda pure se ha qualche dubbio su quanto espresso.

**Comment by marcomeneghin** [comment]
Grazie, quello che non mi è chiaro è cosa succede quando una segnalazione super la data scadenza. Viene cancellata dalla piattaforma e quindi scompare? Non è più modificabile?
Non vorrei dare la possibilità al gestore di cancellare manualmente le segnalazioni ma se non c'è alternativa ragioneremo anche in tal senso.
Grazie, Marco,

**Comment by marcomeneghin** [comment]
Probabilmente ho risolto, rientrando nella piattaforma che utilizzamo per le demo non vedo più le segnalazioni con data scadenza < data odierna. Probabilmente la cancellazione delle segnalzioni scadute è un batch che gira in automatico di notte e la nostra piattaforma per le demo viene accesa solo al bisogno.
Ho interpretato il tutto correttamente? Grazie.

**Comment by evilaliv3** [comment]
Grazie Roberto,


A nostro avviso definire una policy non è così semplice:

1. Log di eccezioni o tentativi di accesso forse sono utili solo per 1 anno
i addirittura 3 mesi per non rischiare di accumulare tante informazioni che
in caso di attacco DoS aprirebbero una facile porta all'attacco

2. Log di azioni operate dagli utenti sul sistema : ha senso che vedano una
policy configurabile

3. Log relativi al trattamento delle segnalazioni secondo noi devono essere
tenuti per tutto il tempo di vita delle segnalazioni relative, e in caso di
cancellazione sotto soglia 2., mantenuti per il tempo definito in 2.

Tecnicamente mi viene immediato pensarne una implementazione, difficile è
chiarire all'utente quanto ho scritto.

@gianlucagilardi: cosa pensi di queste valutazioni?

Ipotizziamo di separare i log in due parti:
1. log di audit relativi alle segnalazioni
2. restanti log di audits

a nostro avviso i log di cui al punto 2. dovrebbero vedere tempo
configurabile (con un default di qpdelle segnalazioni devono essere tenuti
per tutto il tempo di vita della segnalazione ed eventualmente per più
tempo se questa viene
